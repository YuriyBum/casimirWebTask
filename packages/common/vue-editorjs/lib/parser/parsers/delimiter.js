/**
 * Delimiter parser
 * @returns {string}
*/
export const delimiter = () => '<div class="de-delimiter"></div>';
