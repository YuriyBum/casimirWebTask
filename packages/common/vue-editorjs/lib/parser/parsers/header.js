/**
 * Header parser
 * @param {Object} data
 * @param {string} data.text
 * @param {number} data.level
 * @returns {string}
*/
export const header = ({ text, level }) => `<h${level} class="de-header">${text || ''}</h${level}>`;
