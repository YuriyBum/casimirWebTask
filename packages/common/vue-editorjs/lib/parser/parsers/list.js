/**
 * List parser
 * @param {Object} data
 * @param {string} data.style
 * @param {Array} data.items
 * @returns {string}
*/
export const list = ({ style, items }) => {
  const parent = style === 'ordered' ? 'ol' : 'ul';
  const listItems = items
    .map((item) => {
      const itemContent = item instanceof Object ? item.content : item;
      return `<li class="de-list-item">${itemContent}</li>`;
    })
    .join('');

  return `<${parent} class="de-list">${listItems}</${parent}>`;
};
