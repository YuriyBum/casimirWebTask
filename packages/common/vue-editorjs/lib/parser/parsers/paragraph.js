/**
 * Paragraph parser
 * @param {Object} data
 * @param {string} data.text
 * @returns {string}
*/
export const paragraph = ({ text }) => `<p class="de-paragraph">${text || ''}</p>`;
