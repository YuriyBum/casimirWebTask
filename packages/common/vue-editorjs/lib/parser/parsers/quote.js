/**
 * Quote parser
 * @param {Object} data
 * @param {string} data.alignment
 * @param {string} data.text
 * @param {string} data.caption
 * @returns {string}
*/
export const quote = ({ alignment, text, caption }) => {
  const styles = alignment ? `style="text-align: ${alignment}"` : '';
  return `<blockquote ${styles} class="de-quote">
      <p class="de-quote-text">${text || ''}</p>
      <cite class="de-quote-author">${caption || ''}</cite>
    </blockquote>`;
};
