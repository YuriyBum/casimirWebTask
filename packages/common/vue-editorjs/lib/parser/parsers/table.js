const classNames = {
  table: 'de-table',
  head: 'de-head',
  body: 'de-body',
  row: 'de-row',
  cell: 'de-cell'
};

/**
 * Table parser
 * @param {Object} data
 * @param {boolean} data.withHeadings
 * @param {Array.<Array.<string>>} data.content
 * @returns {string}
*/
export const table = ({ withHeadings, content }) => {
  if (!content || content.length === 0) return '';

  let head = '';
  if (withHeadings) {
    const headCells = content[0].map((cellContent) => `<td class="${classNames.cell}">${cellContent}</td>`).join('');
    head = `<thead class="${classNames.head}"><tr class="${classNames.row}">${headCells}</tr></thead>`;
  }

  const bodyRows = content.map((row, index) => {
    if (withHeadings && index === 0) return '';
    const cells = row.map((cellContent) => `<td class="${classNames.cell}">${cellContent}</td>`).join('');
    return `<tr class="${classNames.row}">${cells}</tr>`;
  }).join('');
  const body = `<tbody class="${classNames.body}">${bodyRows}</tbody>`;

  return `<table class="${classNames.table}">${head}${body}</table>`;
};
