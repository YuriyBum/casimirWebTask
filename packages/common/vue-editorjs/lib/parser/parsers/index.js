import { delimiter } from './delimiter';
import { header } from './header';
import { list } from './list';
import { paragraph } from './paragraph';
import { quote } from './quote';
import { table } from './table';

const parsers = {
  delimiter,
  header,
  list,
  paragraph,
  quote,
  table
};

export { parsers };
