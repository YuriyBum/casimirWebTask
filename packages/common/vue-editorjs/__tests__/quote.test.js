// eslint-disable-next-line import/no-extraneous-dependencies
import serializer from 'jest-serializer-html';
import { quote } from '../lib/parser/parsers/quote';

expect.addSnapshotSerializer(serializer);

describe('quote editorjs block parser', () => {
  it('should return empty quote', () => {
    const block = { };
    const result = quote(block);
    expect(result).toMatchSnapshot();
  });

  it('should return quote', () => {
    const block = {
      caption: 'Author',
      text: 'Text'
    };
    const result = quote(block);
    expect(result).toMatchSnapshot();
  });

  it('should return quote with alignment', () => {
    const block = {
      alignment: 'center',
      caption: 'Author',
      text: 'Text'
    };
    const result = quote(block);
    expect(result).toMatchSnapshot();
  });
});
