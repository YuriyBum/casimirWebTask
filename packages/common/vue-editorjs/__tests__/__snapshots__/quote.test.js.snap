// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote editorjs block parser should return empty quote 1`] = `
<blockquote class="de-quote">
  <p class="de-quote-text">
  </p>
  <cite class="de-quote-author">
  </cite>
</blockquote>
`;

exports[`quote editorjs block parser should return quote 1`] = `
<blockquote class="de-quote">
  <p class="de-quote-text">
    Text
  </p>
  <cite class="de-quote-author">
    Author
  </cite>
</blockquote>
`;

exports[`quote editorjs block parser should return quote with alignment 1`] = `
<blockquote style="text-align: center"
            class="de-quote"
>
  <p class="de-quote-text">
    Text
  </p>
  <cite class="de-quote-author">
    Author
  </cite>
</blockquote>
`;
